### PR TITLE
Added unit test for display.set_icon()

### PR DESCRIPTION
**Pull request by amipy**
Added a unit test for pygame.display.set_icon()
This is my first time submitting one of these, let me know if something is wrong so I can try to do it better next time.

### CODE REVIEW

**Comment by MightyJosip** [comment]
This unit test should be part of [this file](https://github.com/pygame/pygame/blob/1476009b05919c96b22ef9abeabc5a623c552ce7/test/display_test.py#L379-L400)

**Review by illume — CHANGES_REQUESTED** [review]
Looks good, just needs to be moved into the display_test.py file. 

(maybe the class could be renamed to TestSetIcon?)

File: test/set_icon_test.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_normal(self):
+        gameIcon = pygame.image.load(example_path("data/alien1.jpg"))
+        self.assertIsNone(pygame.display.set_icon(gameIcon))
+
+    def test_text(self):
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon('testing text')
+
+    def test_integer(self):
```
Comment (illume) [inline test/set_icon_test.py:25]: This test is needed.

File: test/set_icon_test.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon(1234)
+
+    def test_dict(self):
```
Comment (illume) [inline test/set_icon_test.py:29]: This test is needed.

File: test/set_icon_test.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon({'testing text':'test'})
+
+    def test_list(self):
```
Comment (illume) [inline test/set_icon_test.py:33]: This test is needed.

File: test/set_icon_test.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon(['testing text','fdsd'])
+
+    def test_display_not_initialized(self):
+        pygame.display.quit()
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon('test.png')
+
+    def test_audio_not_initialized(self):
```
Comment (illume) [inline test/set_icon_test.py:42]: This test is needed.

File: test/set_icon_test.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    def test_dict(self):
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon({'testing text':'test'})
+
+    def test_list(self):
+        with self.assertRaises(TypeError):
+            pygame.display.set_icon(['testing text','fdsd'])
+
+    def test_display_not_initialized(self):
```
Comment (illume) [inline test/set_icon_test.py:37]: 👍

**Comment by illume** [comment]
There was already a PR open for testing display.set_icon... https://github.com/pygame/pygame/pull/2042
However, I think your test has some value still. The other test does not run by default in CI, only when running locally.

**Comment by illume** [comment]
If you want to continue this, please feel free to comment here. Closing for now to keep our list of PRs clean.

**Pull request closed without merging**